proposals/0000-unicode-for-string-processing.md

```diff
@@ -0,0 +1,1321 @@
+
+```swift
+let data = "\u{300},\u{301},\u{302},\u{303},..."
+for match in data.matches(of: /(.),/.matchingSemantics(.unicodeScalar)) {
+    print(match.1)
+}
+// Prints:
+//  ̀
+//  ́
+//  ̂
+// ...
+```
+
+#### Any grapheme cluster
+
+`Regex` also provides a way to match a single grapheme cluster, regardless of the current semantic level. The **any grapheme cluster** character class is written as `\X` or `CharacterClass.anyGraphemeCluster`, and matches from the current location up to the next grapheme cluster boundary. This includes matching newlines, regardless of any option settings. This metacharacter is equivalent to the regex syntax `(?Xs:.)`.
+
+
+#### Digits
+
+The **decimal digit** character class is matched by `\d` or `CharacterClass.digit`. Both regexes in this example match one or more decimal digits followed by a colon:
+
+```swift
+let regex1 = /\d+:/
+let regex2 = Regex {
+    OneOrMore(.digit)
+    ":"
+}
+```
+
+_Unicode scalar semantics:_ Matches a Unicode scalar that has a `numericType` property equal to `.decimal`. This includes the digits from the ASCII range, from the _Halfwidth and Fullwidth Forms_ Unicode block, as well as digits in some scripts, like `DEVANAGARI DIGIT NINE` (U+096F). This corresponds to the general category `Decimal_Number`.
+
+_Grapheme cluster semantics:_ Matches a character made up of a single Unicode scalar that fits the decimal digit criteria above.
+
+
+To invert the decimal digit character class, use `\D` or `CharacterClass.digit.inverted`.
+
+
+The **hexadecimal digit** character class is matched by  `CharacterClass.hexDigit`.
+
+_Unicode scalar semantics:_ Matches a decimal digit, as described above, or an uppercase or small `A` through `F` from the _Halfwidth and Fullwidth Forms_ Unicode block. Note that this is a broader class than described by the `UnicodeScalar.properties.isHexDigit` property, as that property only include ASCII and fullwidth decimal digits.
+
+_Grapheme cluster semantics:_ Matches a character made up of a single Unicode scalar that fits the hex digit criteria above.
+
+To invert the hexadecimal digit character class, use `CharacterClass.hexDigit.inverted`.
+
+*<details><summary>Rationale</summary>*
+
+Unicode's recommended definition for `\d` is its [numeric type][numerictype] of "Decimal" in contrast to "Digit". It is specifically restricted to sets of ascending contiguously-encoded scalars in a decimal radix positional numeral system. Thus, it excludes "digits" such as superscript numerals from its [definition][derivednumeric] and is a proper subset of `Character.isWholeNumber`. 
+
+We interpret Unicode's definition of the set of scalars, especially its requirement that scalars be encoded in ascending chains, to imply that this class is restricted to scalars which meaningfully encode base-10 digits. Thus, we choose to make the grapheme cluster interpretation *restrictive*.
```

Comment:
But what does "restrictive" even mean? There's no mention of the character properties proposal or how this aligns with it